src/app/containers/OptimizelyArticleCompleteTracking/index.test.jsx

```diff
@@ -0,0 +1,211 @@
+  jest.useFakeTimers();
+  console.error = jest.fn();
+  global.IntersectionObserver = IntersectionObserver;
+});
+
+afterEach(() => {
+  jest.runOnlyPendingTimers();
+  jest.useRealTimers();
+  console.error = error;
+  observers.clear();
+});
+
+describe('Optimizely Page Complete tracking', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return a function that can be assigned to an element to observe for intersections', () => {
+    const { container } = render(
+      <ContextWrap pageType={ARTICLE_PAGE} service="news" isAmp={false}>
+        <OptimizelyArticleCompleteTracking />
+      </ContextWrap>,
+    );
+
+    const element = container.getElementsByTagName('div')[0];
+    const { observe } = getObserverInstance(element);
+
+    expect(observe).toHaveBeenCalledWith(element);
+  });
+
+  it('should not send tracking event when element is not in view', async () => {
+    useOptimizelyVariation.mockReturnValue('variation_1');
+
+    const { container } = render(
+      <ContextWrap pageType={ARTICLE_PAGE} service="news" isAmp={false}>
+        <OptimizelyArticleCompleteTracking />
+      </ContextWrap>,
+    );
+
+    const element = container.getElementsByTagName('div')[0];
+    const observerInstance = getObserverInstance(element);
+
+    act(() => {
+      triggerIntersection({
+        changes: [{ isIntersecting: false }],
+        observer: observerInstance,
+      });
+    });
+
+    await waitFor(() => {
+      expect(optimizely.track).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  it('should not send tracking event when element is in view, but not in experiment variation', async () => {
+    useOptimizelyVariation.mockReturnValue(null);
+
+    const { container } = render(
+      <ContextWrap pageType={ARTICLE_PAGE} service="news" isAmp={false}>
+        <OptimizelyArticleCompleteTracking />
+      </ContextWrap>,
+    );
+
+    const element = container.getElementsByTagName('div')[0];
+    const observerInstance = getObserverInstance(element);
+
+    act(() => {
+      triggerIntersection({
+        changes: [{ isIntersecting: true }],
+        observer: observerInstance,
+      });
+    });
+
+    await waitFor(() => {
+      expect(optimizely.track).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  it('should not send tracking event when element is in view, but is on AMP', async () => {
+    useOptimizelyVariation.mockReturnValue('variation_1');
+
+    const { container } = render(
+      <ContextWrap pageType={ARTICLE_PAGE} service="news" isAmp>
+        <OptimizelyArticleCompleteTracking />
+      </ContextWrap>,
+    );
+
+    const element = container.getElementsByTagName('div')[0];
+    const observerInstance = getObserverInstance(element);
+
+    act(() => {
+      triggerIntersection({
+        changes: [{ isIntersecting: true }],
+        observer: observerInstance,
+      });
+    });
+
+    await waitFor(() => {
+      expect(global.IntersectionObserver).toHaveBeenCalledTimes(1);
```

Comment:
In a way you wouldn't expect this to be called on amp as no user js is allowed to run. You might want to go a step further and return null from the the component on amp rather than rendering the div; you could test for this